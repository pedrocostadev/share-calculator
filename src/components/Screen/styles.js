import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    alignItems: 'flex-end',
    justifyContent: 'center',
    backgroundColor: '#141414',
    height: '100%',
  },
});

export default styles;
