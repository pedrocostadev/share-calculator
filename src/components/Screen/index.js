import React from 'react';
import PropTypes from 'prop-types';
import { View, ScrollView } from 'react-native';

import styles from './styles';

const Screen = ({ children }) => (
  <View style={styles.container}>
    <ScrollView>{children}</ScrollView>
  </View>
);

Screen.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Screen;
