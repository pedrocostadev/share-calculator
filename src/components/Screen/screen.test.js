import React from 'react';
import { Text } from 'react-native';
import { render, cleanup } from '@testing-library/react-native';

import Screen from './';

describe('<Screen />', () => {
  afterEach(cleanup);

  test('should render children', async () => {
    const text = 'test-the-screen';

    const { queryAllByText, container } = render(
      <Screen>
        <Text>{text}</Text>
      </Screen>,
    );
    const ocurrences = await queryAllByText(text);
    expect(ocurrences).toHaveLength(1);
    expect(container).toMatchSnapshot();
  });
});
