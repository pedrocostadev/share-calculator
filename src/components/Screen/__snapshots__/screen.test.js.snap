// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Screen /> should render children 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "#141414",
        "display": "flex",
        "height": "100%",
        "justifyContent": "center",
      }
    }
  >
    <ScrollView>
      <Text>
        test-the-screen
      </Text>
    </ScrollView>
  </View>
</View>
`;
